src/extent.c

```diff
@@ -2151,9 +2148,23 @@ extent_split_wrapper(tsdn_t *tsdn, arena_t *arena,
 }
 
 static bool
-extent_merge_default_impl(void *addr_a, void *addr_b) {
+extent_merge_default_impl(extent_t *a, void *addr_a, extent_t *b,
+    void *addr_b) {
 	if (!maps_coalesce) {
-		return true;
+		/*
+		 * When coalesce is not always allowed (Windows), only merge
+		 * extents with identical sn (i.e. from the same VirtualAlloc
```

Comment:
Can you include some back-of-the envelope math to convince me / the reader that "same sn == same VirtualAlloc call"? This is the first time that we're making that fact a correctness issue instead of just a QoI one, and that makes me somewhat nervous.